WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -278,6 +278,20 @@ def update_events
     end
   end
 
+  def update_wcif_events
+    @competition = competition_from_params
+    wcif_events = params["_json"].map { |wcif_event| wcif_event.permit!.to_h } #<<< TODO: validate that the incoming stuff is actually wcif >>>
```

Comment:
@larspetrus, here's where we blindly assume that incoming json is a valid wcif document. I'm now convinced that we need to validate the schema of the incoming JSON. Do you have any ideas for how to do that? Some light googling led me to https://github.com/ruby-json-schema/json-schema.